### ISSUE TITLE
Ensemble Model interpretability

### ISSUE THREAD

**Issue by rafaloz** [shap/shap#2864]
Hello, 

I have a doubt regarding the use of SHAP for an ensemble model. 

I trained a model on 10-fold cross-validation and I am using the stack of trained models as an ensemble for my target. I use MLPs as the base model. For each fold, I perform standardization of my features and feature selection selecting a subset of 40 features with which I train the fold's MLP. Therefore, at the end of the training, I would have an ensemble model with 400 input features (40 features x10 MLPs). Even though among these features some are repeated (I have checked it and at the end of the day I have around 100 different features) due to the standardization step within the 10-fold cv, a different training set is used on every fold, there is a bit of variability among the features.  

To interpret the whole model (ensemble 10 MLPs) I thought about obtaining the explainer of each MLP using its own training set and then obtaining the SHAP values for each of the MLPs of my test set. Then I would sum up the SHAP values of the repeated features, even though these are not exactly the same due to the standardization and I would use as the expected value the average of the sum of the expected values of each MLP. Since for building the ensemble I am just averaging the final results of the MLPs and the SHAP explainer model is linear, I think I am not violating any mathematical conditions. 

Would this be rigorous? are the SHAP values of an ensemble model (average) the average of the SHAP values of each model of which it is composed? 

I am obtaining the SHAP values using Kernel SHAP. 

Thanks for the library and your patience!

**Comment by Jose-Verdu-Diaz** [comment]
Slightly unrelated, but this issue might be of some help: #112

Adding or averaging the shape values of an ensemble of models taking different features and features with different distributions seems unreliable. An approach would be analyzing the shape values of each model of the ensemble separately. This would, of course, make model interpretability harder than having a unique XAI analysis.
